plugins/modules/aci_dhcp_relay_provider.py

```diff
@@ -0,0 +1,357 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# Copyright: (c) <year>, <Name> (@<github id>)
+# GNU General Public License v3.0+ (see LICENSE or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+DOCUMENTATION = r'''
+---
+module: aci_dhcp_relay_provider
+short_description: Manage DHCP relay policy providers.
+description:
+- Manage DHCP relay policy providers (dhcpRsProv) configuration on Cisco ACI fabrics.
+options:
+  tenant:
+    description:
+    - Name of an existing tenant.
+    type: str
+  relay_policy:
+    description:
+    - Name of an existing DHCP rely policy
+    type: str
+    aliases: [ relay_policy_name ]
+  epg_type:
+    description:
+    - Type of EPG the DHCP srver is in
+    type: str
+    choices: [ app_epg, l2_external_net, l3_external_net ]
+    required: yes
+  anp:
+    description:
+    - Application Profile the EPG is in.
+    - Only used when epg_type is app_epg.
+    type: str
+  app_epg:
+    description:
+    - Name of the Application EPG the DHCP server is in.
+    - Only used when epg_type is app_epg
+    type: str
+  l2out_name:
+    description:
+    - Name of the L2out the DHCP server is in.
+    - Only used when epg_type is l2_external_net
+    type: str
+  l3out_name:
+    description:
+    - Name of the L3out the DHCP server is in.
+    - Only used when epg_type is l3_external_net.
+    type: str
+  external_net:
+    description:
+    - Name of the external network object the DHCP server is in.
+    - Only used when epg_type is l2_external_net or l3_external_net.
+    type: str
+  dhcp_server_addr:
+    description:
+    - DHCP server address
+    type: str
```

Comment:
We seems to be missing the tenant in which the provider is located. We have the tenant of the policy but not the one for the dhcp server as they can be in different tenants.

Reply:
Added a provider_tenant var.